File: backend/app/routers/inbox.py
from fastapi import APIRouter, HTTPException
from app.database import get_db
from app.repositories.inbox_repo import InboxRepo
from app.models.inbox import InboxCreate, InboxItem

router = APIRouter(prefix="/api/inbox", tags=["inbox"])


@router.post("", response_model=InboxItem)
async def create_inbox(item: InboxCreate):
    db = await get_db()
    try:
        repo = InboxRepo(db)
        result = await repo.create(item.content)
        return result
    finally:
        await db.close()


@router.get("", response_model=list[InboxItem])
async def get_pending_inbox(date: str | None = None):
    db = await get_db()
    try:
        repo = InboxRepo(db)
        return await repo.get_pending(date)
    finally:
        await db.close()


@router.delete("/{inbox_id}")
async def delete_inbox(inbox_id: int):
    db = await get_db()
    try:
        repo = InboxRepo(db)
        deleted = await repo.delete(inbox_id)
        if not deleted:
            raise HTTPException(status_code=404, detail="碎片不存在")
        return {"ok": True}
    finally:
        await db.close()

File: frontend/src/api/inbox.ts
import client from './client'

export interface InboxItem {
  id: number
  content: string
  created_at: string
  status: string
  processed_date: string | null
}

export const inboxApi = {
  create: (content: string) => client.post<unknown, InboxItem>('/inbox', { content }),
  getPending: (date?: string) => client.get<unknown, InboxItem[]>('/inbox', { params: { date } }),
  delete: (id: number) => client.delete(`/inbox/${id}`),
}
